src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -45,20 +45,20 @@ public static string HtmlEncode(string value)
                 return value;
             }
 
-            StringBuilder sb = StringBuilderCache.Acquire(value.Length);
-            HtmlEncode(value, index, sb);
-            return StringBuilderCache.GetStringAndRelease(sb);
+            Span<char> charSpan = stackalloc char[value.Length];
```

Comment:
This is dangerous.  value could be really big, in which case we'd try to stackalloc that much space and could easily result in a stack overflow / crash.  This should instead just using a fixed amount, e.g. 256, or not use stack space at all and just pass an empty span to ValueStringBuilder.  Or update ValueStringBuilder with a capacity constructor.

Note, though, that @benjamin-hodgson already has a PR that addresses WebUtility at https://github.com/dotnet/corefx/pull/27250, so it probably makes sense to just undo these WebUtility.cs changes and let his PR handle that.

Reply:
For the sake of this PR I would recommend just to pass a `stackalloc char[255]` to the ValueStringBuilder and let @benjamin-hodgson fine grain the stack allocation.

Reply:
@ViktorHofer, that makes me nervous, given that these changes showed a regression on throughput in his PR: https://github.com/dotnet/corefx/pull/27250#issuecomment-392105024.

Reply:
I see, that's unfortunate, I didn't know about the regression.

Reply:
So, do you like me to undo WebUtility changes?

Reply:
Yes, but you also need to undo the removal of the StringBuilderCache from the assembly.

Reply:
of course: done